### PR TITLE
page_foreground output to hdf summary file

### PR DESCRIPTION
**Pull request by GarethDaviesGW**
This is very similar to #3767, but specifically for the v1.16  branch

There are a couple of minor content changes compared to that PR, as the output now gives the trigger information for every column in the trigger_merge file (as appropriate), as well as the template duration

### CODE REVIEW

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
...
+        mchirp, _ = mass1_mass2_to_mchirp_eta(mass1, mass2)
+
+        ofd.create_dataset('chirp_mass', data=mchirp, dtype=np.float32)
+
+        sngl_fields = [f for f in self.sngl_files[self.ifos[0]].columns
+                       if not ('template' in f)]
+        # Single-detector fields
+        logging.info("Outputting single-detector info")
+        for field in sngl_fields:
+            if field in ['gating','search','snr']: continue
```
Comment (GarethDaviesGW) [inline pycbc/io/hdf.py:1041]: This bit is the main difference from the other PR, as we get all of the single-detector columns from the trigger_merge files and then filter out the ones that we don't want

File: pycbc/io/hdf.py [hunk 2]
```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
...
+
+        logging.info("Outputting template duration")
+        temp_dur_vals_valid = self.get_snglfile_array_dict('template_duration')
+        temp_dur = np.zeros_like(snr_vals_valid[self.ifos[0]][0])
+        for ifo in self.ifos:
+            vals = temp_dur_vals_valid[ifo][0]
+            valid = temp_dur_vals_valid[ifo][1]
+            temp_dur[valid] = vals[valid]
+        ofd.create_dataset('template_duration', data=temp_dur, dtype=np.float32)
+        logging.info(all(temp_dur > 0))
```
Comment (GarethDaviesGW) [inline pycbc/io/hdf.py:1073]: This is also a change from the previous PR, as we are getting the template duration. This is a slight faff, as we cant just use one of the ifos as each event wont have triggered in every one, so we have just overwrite this each time

File: pycbc/io/hdf.py [hunk 3]
```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
...
+        ofd.close()
```
Comment (GarethDaviesGW) [inline pycbc/io/hdf.py:1074]:
```suggestion
        ofd.close()
```

File: pycbc/io/hdf.py [hunk 4]
```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
...
+
+        ofd.create_dataset('chirp_mass', data=mchirp, dtype=np.float32)
+
+        sngl_fields = [f for f in self.sngl_files[self.ifos[0]].columns
+                       if not ('template' in f)]
+        # Single-detector fields
+        logging.info("Outputting single-detector info")
+        for field in sngl_fields:
+            if field in ['gating','search','snr']: continue
+            logging.info(field)
```
Comment (GarethDaviesGW) [inline pycbc/io/hdf.py:1042]:
```suggestion
```

**Comment by spxiwh** [comment]
This is a merge into a release branch and won't appear on master, so I'll trust your judgement on this one, as no-one but you should end up using these changes!

**Review by spxiwh — APPROVED** [review]

**Pull request merged**